simulators/ethereum/pyspec/runner.go

```diff
@@ -0,0 +1,207 @@
+// creates a testcase struct for each test, and passes the testcase struct to a    //
+// func() parameter `fn`, which is used within fixtureRunner() to run the tests.   //
+// ------------------------------------------------------------------------------- //
+func loadFixtureTests(t *hivesim.T, root string, fn func(testcase)) {
+	filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		// check file is actually a fixture
+		if err != nil {
+			t.Logf("unable to walk path: %s", err)
+			return err
+		}
+		if info.IsDir() || !strings.HasSuffix(info.Name(), "withdrawals_use_value_in_tx.json") {
+			return nil
+		}
+
+		// extract fixture.json tests (multiple forks) into fixtureTest structs
+		var fixtureTests map[string]fixtureTest
+		if err := common.LoadJSON(path, &fixtureTests); err != nil {
+			t.Logf("invalid test file: %v, unable to load json", err)
+			return nil
+		}
+
+		// create testcase structure from fixtureTests
+		for name, fixture := range fixtureTests {
+
+			if !strings.HasSuffix(name, "000_shanghai") {
+				continue
+			}
+
+			// skip networks post merge or not supported
+			network := fixture.json.Network
+			for _, skip := range []string{"Istanbul", "Berlin", "London"} {
+				if strings.Contains(network, skip) || envForks[network] == nil {
+					t.Logf("skipping test for network %s", network)
+					continue
+				}
+			}
+
+			// define testcase (tc) struct with initial fields
+			tc := testcase{
+				fixture:  fixture,
+				name:     name,
+				filepath: path,
+			}
+
+			// extract genesis, payloads & post allocation field to tc
+			tc.extractFixtureFields(fixture.json)
+
+			// feed tc to single worker within fixtureRunner()
+			fn(tc)
+		}
+		return nil
+	})
+}
+
+// ----------------------------------------------------------------------------------------------------//
+// run() executes a testcase against the client, called within a test channel from fixtureRunner().    //
+// All testcase payloads are sent and executed using the EngineAPI. For verification all fixture       //
+// nonce, balance and storage values are checked against the response recieved from the lastest block. //
+// ----------------------------------------------------------------------------------------------------//
+func (tc *testcase) run(t *hivesim.T) {
+	start := time.Now()
+
+	t.Log("HIVE LOG --> Setting variables required for starting client.")
+	engineAPI := hive_rpc.HiveRPCEngineStarter{
+		ClientType: tc.clientType,
+		EnginePort: globals.EnginePortHTTP,
+		EthPort:    globals.EthPortHTTP,
+		JWTSecret:  globals.DefaultJwtTokenSecretBytes,
+	}
+	ctx := context.Background()
+	env := hivesim.Params{
+		"HIVE_FORK_DAO_VOTE": "1",
+		"HIVE_CHAIN_ID":      "1",
+		"HIVE_SKIP_POW":      "1",
+		"HIVE_NODETYPE":      "full",
+	}
+	tc.updateEnv(env)
+	t0 := time.Now()
+
+	// start client (also creates an engine RPC client internally)
+	t.Log("HIVE LOG --> Starting client with Engine API.")
```

Comment:
Please remove these `HIVE LOG -->`, `HIVE FATAL -->` etc. prefixes. It is sufficient to prefix the error message in `t.Fatalf()` calls with simply `Error:`

Reply:
Sure, happy to remove! I added it to help distinguish where the log/error messages were coming from (i.e hive or geth). But I've only just realised now that the differing containers achieve this :)

Reply:
BTW, you can also run the web interface locally to check your results.  See here for more information: https://github.com/ethereum/hive/blob/master/docs/commandline.md#viewing-simulation-results-hiveview